### ISSUE TITLE
Upgrade crap user safeties

### ISSUE THREAD

**Issue by bethlakshmi** [bethlakshmi/gbe-divio-djangocms-python2.7#193]
put in a new google captcha 
put in the ability to black list an email address
consider a bulk delete feature in the user management page

**Comment by bethlakshmi** [comment]
We have 4 new cases since the writing of this ticket (at least).
Also can conjoin this with the work to fix case sensitivity on email redundancy check (Guilted Lili's user issue)

**Comment by bethlakshmi** [comment]
OK, here's my plan:
- I found a repository - https://www.stopforumspam.com/usage - which keeps a record of users seen misbehaving.  All of the bad user emails that I tested are found here (tested 6 out of 9 or so). 
- I wrote code to call the repository API and just flatly deny users who appear on this list.  I could do nuances like "last appeared" date, or frequency of reports - but I don't really have a clue what nuances are worth it.  I figure I can make it more complicated later.
- Make a customizable message for the error.
- Also do the "do we already have this email?" check as case insensitive.  Make that message customizable too.

I could do more, here, but i figured it made sense to do small steps.  Other future options:
- keep our own cache of email domains - given that we see lots of repeats from certain problem domains (the repo doesn't really seem to have a domain check)
- age off the bad reports in case someone's email was harvested - like if it hasn't been reported in 90 days, it's on the "OK" list again
- make a more complicated "are you a human?" check - but honestly, I think this is actual humans, so a more complicated check will be more complicated for our users and do very little.
